### DISCUSSION TITLE
Assignment Repositories from Forks - Read Role Issue

### DISCUSSION THREAD

**Discussion by CakeJamble** [community/community#110773] [GitHub Education]
### Hi everyone,

# Problem
There are some privacy issues with trying to create an assignment with GitHub Classroom from a private repository.
1. If an assignment repository is private, and then distributed to students as a fork of the assignment repository, students cannot sync changes from upstream, because the assignment repository is not visible. This means you have to make your assignments public to ensure that students can sync upstream, which may be undesirable for a number of reasons.
2. If you try to keep the assignment private, and add a student as a collaborator with ONLY the Read role, they will be able to sync changes from upstream. However, they are also automatically granted the Read role to all forks of the assignment repository. This means that they can suddenly see the work of all fellow classmates.

**Comment by guyer** [comment]
This is not quite what I observe, but close enough.

In my case, I have a private starter repository and, when they accept an assignment, students do get a private fork. The problem is that they are then granted **Read** access to the starter repository, which means that they can see any pull requests made by other students against the starter repository.

I have instructed students to make PRs in their own fork, but there appears to be nothing enforcing them to do so.

I can remove permissions automatically granted to the starter repository, but that seems to defeat the purpose of making everything a fork. Would students still be able to **Sync fork**?

**Comment by adaskin** [comment]
+ in team-assignments, it adds team members(students) as organization members and grant access to the teams  based on the organization default settings, which was "read" in my case.  
So students were able to read all the forks created before them. After being informed by students,  I had to spend two hours to remove granted accesses from each teams.

**Comment by sgoggins** [comment]
Also experiencing this issue.
